src/remember.ts

```diff
@@ -0,0 +1,55 @@
+import { Operator, StrictSink } from './types';
+
+export function remember<T>(numRemembered: number = 1): Operator<T, T> {
+  return source => {
+    const marker = {};
+    let sinks: Array<StrictSink<T> | undefined> = [];
+    let lasts: T[] = Array(numRemembered).fill(marker);
+    let curr = -1;
+    let started = false;
+
+    const mkTalkback = (sink: StrictSink<T>) => () => {
+      sinks[sinks.indexOf(sink)] = void 0;
+    };
+
+    return (_, sink) => {
+      sinks.push(sink);
+
+      if (!started) {
+        source(0, (t, d) => {
```

Comment:
So it seems like the source is never unsubscribed. This may or may not be a problem, I don't have a strong opinion about it, but I think we could talk about it, like at least trying to find use cases where never-unsubscribing will be problematic.

This is also different to xstream `remember()` because xstream MemoryStreams are unsubscribed when they complete, and in the 1st test you wrote, xstream remember() would actually cause `numHook` to be 2. This is both an advantage and a problem in xstream. It's an advantage because the unsubscribe happens automatically, so by default it has better protection against memory leaks. It's a problem because it's a stumbling stone, people constantly have problems with this case, and I myself have created a silly operator to force the unsubscribe to never happen https://github.com/staltz/manyverse/blob/304752070876bd4e517897c11a46c9b054cd513a/src/frontend/screens/conversation/model.ts#L44-L46

As I said, I am not sure what to do here. On one hand, this new `remember()` prevents the gotcha and will make it easier for people to use, but on the other hand it doesn't give you any way of unsubscribing if you would actually want to do that. At least the xstream `remember()` gives you a way of opting-in to "never-unsubscribe" with the `dropCompletion` operator. Could you figure out a way of opting-in to "unsubscribe as soon as it completes" for this one?

Reply:
I've added an option now that let's the user unsubscribe if all listeners have unsubscribed.